### PR TITLE
Contributing to vHive document

### PR DESCRIPTION
**Pull request by MBaczun**
Adds a page with guidelines on contributing to vHive.

### CODE REVIEW

**Review by ustiugov — CHANGES_REQUESTED** [review]
Very good, thanks! Please add a reference to this doc in the main readme file.

**Comment by ustiugov** [comment]
also, please fix the linter check

File: docs/contributing_to_vhive.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+4. Merge the upstream changes:
+	```bash
+	git merge upstream/main
+	```
+
+You can check out [the official GitHub docs](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) for more information on syncing forks.
+
+### Squashing Commits
+We prefer for every commit in the repo to encapsulate a single concrete and atomic change/addition. This means our commit history shows a clear and structured progression, and that it remains consise and readable. In your own branch you can clean up your commit history by squashing smaller commits together using `git rebase`:
+1. Find the hash of the oldest commit which you want to squash up to. For example, if you made three commits in sequence (A, B, C, such that C is the latest commit) and you wanted to squash A and B together, you would need to find the hash of A. You can find the hash of a commit on GitHub or by using the command:
```
Comment (boramalper) [inline docs/contributing_to_vhive.md:37]: For three commits A < B < C in time, if I want to squash A and B together, I need the hash of "A-1". The correct example would be squashing B and C (and finding the hash of A). Indeed reading further, this is I believe what you meant.

File: docs/contributing_to_vhive.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Then we are interested in rebasing such that the tree ends up like this:
+```
+              A'--B'--C' branch
+             /
+D---E---F---G master
+```
+For more detail check out [the git rebase documentation](https://git-scm.com/docs/git-rebase).
+
+To rebase on main simply checkout your branch and use the following:
+	```bash
```
Comment (boramalper) [inline docs/contributing_to_vhive.md:77]: Spurious indentation?

**Review by boramalper — CHANGES_REQUESTED** [review]
LGTM too =) Two minor changes requested.

File: docs/contributing_to_vhive.md (hunk 1 above)
Comment (MBaczun) [reply docs/contributing_to_vhive.md]: thanks for the correction, resolved.

**Pull request merged**